core/app/models/spree/variant/price_selector.rb

```diff
@@ -24,13 +24,28 @@ def initialize(variant)
 
       # The variant's price, given a set of pricing options
       # @param [Spree::Variant::PricingOptions] price_options Pricing Options to abide by
-      # @return [Spree::Money, nil] The most specific price for this set of pricing options.
-      def price_for(price_options)
-        variant.currently_valid_prices.detect do |price|
+      # @param [Boolean] return_price Flag to return a Spree::Price instead of
+      #   a Spree::Money
+      # @return [Spree::Price, Spree::Money, nil] The most specific price for
+      #   this set of pricing options in the desired format.
+      def price_for(price_options, return_price: false)
```

Comment:
I'm a bit concerned by this approach to deprecating the method. Our approach to major upgrades is to ensure that someone who's on the latest 3.x will always be able to update to 4.0 _without touching their code_ as long as they don't get any deprecation warnings.

Unless I'm missing something, with this change people will have to call `price_for(..., return_price: true)` in the latest 3.x in order not to get the deprecation warning. But then, in 4.0, we are going to remove the `return_price:` argument altogether, and anyone who passes it will start getting an error.

In other words, people who upgrade from the latest 3.x to 4.0 will have to change something in their code in order to complete the upgrade, which is something we should try to avoid if at all possible.

How about removing this, and only using the value of `Spree::Config. price_selector_returns_price` instead? It's also more straightforward.

Reply:
The idea was to deprecated `return_price` after the change in the next major (4.0). This kind of deprecation allows keeping the system working for example if the host application switched to returning price but some extensions don't.

I'm also ok using `Spree::Config. price_selector_returns_price` alone.

Reply:
Interesting, I'd not really considered this. So @kennyadsl are you thinking we do this:

- In 3.x you get a deprecation warning if you use the `return_price: false` behaviour, but the behaviour remains.
- In 4.x we remove the `return_price: false` behaviour and you get a deprecation warning if you pass `return_price:` to the method.
- In 5.x we remove the `return_price:` argument.

That's not unreasonable, but 5.0 seems so far in the distance. 😅

Reply:
That's what I was thinking but, totally unrelated, I think we can change the removal deprecation process starting from Solidus 3.0 so that we are allowed to remove deprecated code in the next minor version. That's what Rails does, and also (I discovered this yesterday 😓) what we kinda state [here](https://guides.solidus.io/developers/upgrades/versioning-guidelines.html):

> We would like to also follow the Rails approach: deprecating functionality in one minor version and removing it in the next.

Anyway, I think @aldesantis has a valid proposal to avoid this parameter at all, we were discussing it yesterday and I guess he will post some comments soon.